### ISSUE TITLE
No gradient through the cnn encoder

### ISSUE THREAD

**Issue by FaisalAhmed0** [rll-research/cic#5]
In CIC code the observations are encoded inside torch.no_grad which means that there is no gradient calculation for the CNN, which means that the CNN weights will not be updated during pre-training.
https://github.com/rll-research/cic/blob/b523c3884256346cb585bf06e52a7aadc127dcfc/agent/cic.py#L216
